### PR TITLE
feat: use the pre-commit hooks library of apm-pipeline-library

### PR DESCRIPTION
**Pull request by v1v**
## Highlights
- Let's consume the pre-commit hooks library.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -36,52 +36,15 @@ repos:
         types: [yaml]
         exclude: (^.pre-commit-config.yaml$|^.ci/.yamlint.yml$|^target/)
 
--   repo: local
+-   repo: git@github.com:v1v/apm-pipeline-library
```
Comment (v1v) [inline .pre-commit-config.yaml]: This should be changed once the feature has been merged in this repo

File: .ci/Jenkinsfile [hunk 2]
```diff
@@ -78,8 +78,10 @@ pipeline {
             withGithubNotify(context: 'Sanity checks', tab: 'tests') {
               deleteDir()
               unstash 'source'
-              dir("${BASE_DIR}"){
-                preCommit(commit: "${GIT_BASE_COMMIT}", junit: true)
+              dir(BASE_DIR){
+                sshagent(['f6c7695a-671e-4f4f-a331-acdce44ff9ba']) {
```
Comment (mdelapenya) [inline .ci/Jenkinsfile]: why running the preCommit in an SSH agent?

Comment (v1v) [reply .ci/Jenkinsfile]: sshagent is required as long as https://github.com/elastic/apm-integration-testing/pull/571/files#diff-82c3b36709926de3a1b27bea0d8258ccR39 is a private repo.

Not sure yet whether those implementation details should be defined within the `preCommit` step to keep the pipeline simple or just wait until the private-repo has been set.  That's the reason I have not set ready this PR yet, but wanted to PoC this approach worked in other repos.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (v1v) [reply .pre-commit-config.yaml]:
```suggestion
-   repo: git@github.com:elastic/apm-pipeline-library
```

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -36,52 +36,15 @@ repos:
...
+    rev: feature/precommit-lib
```
Comment (v1v) [inline .pre-commit-config.yaml]:
```suggestion
    rev: current
```

File: .ci/Jenkinsfile (hunk 2 above)
Comment (v1v) [reply .ci/Jenkinsfile]: Already removed from the pipeline itself, as it's now provided within the `preCommit` step then once we open the library we can deprecate that particular behavior without changing any pipelines.

**Pull request merged**